envoy/network/socket.h

```diff
@@ -92,6 +93,18 @@ class SocketAddressProvider {
    * @param indent_level the level of indentation.
    */
   virtual void dumpState(std::ostream& os, int indent_level) const PURE;
+
+  /**
+   * @return the upstream SSL connection. This will be nullptr if the upstream
+   * connection does not use SSL.
+   */
+  virtual Ssl::ConnectionInfoConstSharedPtr upstreamSslConnection() const PURE;
+
+  /**
+   * @return the downstream SSL connection. This will be nullptr if the downstream
+   * connection does not use SSL.
+   */
+  virtual Ssl::ConnectionInfoConstSharedPtr downstreamSslConnection() const PURE;
```

Comment:
This can be just `sslConnection()` after removing `upstreamSslConnection()`

Reply:
got it, thanks